cleanlab/regression/learn.py

```diff
@@ -0,0 +1,850 @@
+
+.. code:: python
+
+    from sklearn.base import BaseEstimator
+
+    class YourModel(BaseEstimator):
+        def __init__(self, ):
+            pass
+        def fit(self, X, y, sample_weight=None):
+            pass
+        def predict(self, X):
+            pass
+        def score(self, X, y, sample_weight=None):
+            pass
+            
+"""
+
+from typing import Optional, Union, Tuple
+import inspect
+import warnings
+
+import math
+import numpy as np
+import pandas as pd
+
+import sklearn.base
+from sklearn.base import BaseEstimator
+from sklearn.model_selection import KFold
+from sklearn.linear_model import LinearRegression
+from sklearn.metrics import r2_score
+
+from cleanlab.typing import LabelLike
+from cleanlab.internal.constants import TINY_VALUE
+from cleanlab.internal.util import train_val_split, subset_X_y
+from cleanlab.internal.regression_utils import assert_valid_regression_inputs
+from cleanlab.internal.validation import labels_to_array
+
+
+class CleanLearning(BaseEstimator):
+    """
+    CleanLearning = Machine Learning with cleaned data (even when training on messy, error-ridden data).
+
+    Automated and robust learning with noisy labels using any dataset and any model. This class
+    trains a ``model`` with error-prone, noisy labels as if the model had been instead trained
+    on a dataset with perfect labels. It achieves this by cleaning out the error and providing
+    cleaned data while training. This class is currently intended for standard regression tasks.
+
+    Parameters
+    ----------
+    model :
+        A model implementing the `sklearn estimator API <https://scikit-learn.org/stable/developers/develop.html#rolling-your-own-estimator>`_,
+        defining the following functions:
+
+        - ``model.fit(X, y, sample_weight=None)``
+        - ``model.predict(X)``
+        - ``model.score(X, y, sample_weight=None)``
+
+        If the model is not sklearn-compatible by default, it might be the case that
+        standard packages can adapt the model. For example, you can adapt PyTorch
+        models using `skorch <https://skorch.readthedocs.io/>`_ and adapt Keras models
+        using `SciKeras <https://www.adriangb.com/scikeras/>`_.
+
+        Stores the model used in CleanLearning.
+        Default model used is `sklearn.linear_model.LinearRegression
+        <https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html>`_.
+
+    cv_n_folds :
+        This class needs holdout predictions for every data example and if not provided,
+        uses cross-validation to compute them. This argument sets the number of cross-validation
+        folds used to compute out-of-sample predictions for each example in ``X``. Default 5.
+
+    n_boot :
+        Number of bootstrap samplings iterations that should be conducted to estimate the model's
+        epistemic uncertainty. Default 5.
```

Comment:
```suggestion
        Number of bootstrap resampling rounds used to estimate the model's epistemic uncertainty.
        Default is 5. Higher values are expected to produce better results but require longer runtimes.
```

Reply:
Can we allow `n_boot = 0`, in which case we just hardcode the epistemic uncertainty = 0?

Reply:
can we also add an optional boolean argument: `include_aleatoric_uncertainty` which if False, just hardcodes the aleatoric uncertainty = 0 (default is True)